design/ironic-standalone-networking-service.md

```diff
@@ -0,0 +1,739 @@
+  address: 192.168.1.10
+  credentials:
+    type: "password"
+    secretName: switch-credentials
+  macAddress: 3e:c5:c0:cb:fa:f6
+```
+
+Metal³ automatically generates the Ironic Networking switch configuration file
+and restarts the service when switches are added or modified.
+
+The credentials Secret is expected to contain keys for the `username`,
+`password`, and (optionally) `enable-secret`.
+
+#### Story 2
+
+As a Metal³ user, I want to define network configurations for my bare metal
+hosts declaratively so that switch ports are automatically configured when nodes
+are provisioned.
+
+The user creates a `HostNetworkAttachment` defining the desired network
+configuration for an access mode port for the primary network interface:
+
+```yaml
+apiVersion: metal3.io/v1alpha1
+kind: HostNetworkAttachment
+metadata:
+  name: machine-network-VLAN100
+  namespace: some-cluster-namespace
+spec:
+  mode: access
+  nativeVLAN: 100
+```
+
+The user optionally creates other `HostNetworkAttachment` instances
+defining the desired network configuration for trunk ports for the secondary
+network interface(s):
+
+```yaml
+apiVersion: metal3.io/v1alpha1
+kind: HostNetworkAttachment
+metadata:
+  name: data-network-VLANs-200-202
+  namespace: some-cluster-namespace
+spec:
+  mode: trunk
+  allowedVLANs: [200, 201, 202]
+```
+
+Then references them in the `BareMetalHost`:
+
+```yaml
+apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+metadata:
+  name: worker-1
+  namespace: some-host-namespace
+spec:
+  # ... other fields ...
+  networkInterfaces:
+  - macAddress: "aa:bb:cc:dd:ee:01"
+    networkAttachment:
+      namespace: some-cluster-namespace
+      name: machine-network-VLAN100
+  - macAddress: "aa:bb:cc:dd:ee:02"
+    networkAttachment:
+      namespace: some-cluster-namespace
+      name: data-network-VLANs-200-202
+```
+
+When the host is provisioned, Metal³ automatically configures the corresponding
+switch port for the primary interface to access mode on VLAN 100, and the
+secondary interface to trunk mode with tagged VLANs 200-202.
+
+Ideally, `HostNetworkAttachment` instances should only be referenced by
+`BareMetalHost` instances after initial inspection so that NIC information is
+present in the `HardwareData` object and available for validation of the
+network interface info by a validation webhook. If inspection has not yet run
+the validation webhook _could_ defer validation until the end of the initial
+inspection phase.
+
+#### Story 3
+
+As a Metal³ operator, I want to configure a service network VLAN that allows
+bare metal hosts to communicate with Ironic during inspection and provisioning,
+separate from their final tenant network configuration.
+
+The operator configures the service network in the Ironic resource:
+
+```yaml
+apiVersion: metal3.io/v1alpha1
+kind: Ironic
+metadata:
+  name: default
+  namespace: default-ironic-namespace
+spec:
+  networkingService:
+    enabled: true
+    networkDriver: standalone-experimental
```

Comment:
I'm a bit confused about the meaning of this field. Isn't it the Node's network interface? Why is it a part of the networking service configuration?